### PR TITLE
Helpfilter: Fix crash in suggestions

### PR DESCRIPTION
**Pull request by mia-pi-git**

### CODE REVIEW

File: server/chat-plugins/help.ts [hunk 1]
```diff
@@ -142,7 +142,7 @@ export class HelpResponder {
 		return null;
 	}
 	getFaqID(faq: string) {
```
Comment (Zarel) [inline server/chat-plugins/help.ts:144]:
```suggestion
	getFaqID(faq: string | undefined) {
```

It seems like this signature is wrong? And/or some function that calls it is wrong?

Comment (mia-pi-git) [reply server/chat-plugins/help.ts:144]: It's not _supposed_ to be undefined, but apparently it could be, and this is a cleaner fix, i felt.

**Comment by Zarel** [comment]
There's still a type signature further up the call stack which is wrong and needs to be corrected, unfortunately.

**Comment by mia-pi-git** [comment]
> There's still a type signature further up the call stack which is wrong and needs to be corrected, unfortunately.

Actually, based on the crash, there isn't. `getFaqID` is directly called from the offending command, where I mistakenly forgot to assert FAQ was defined. This corrects that in a cleaner way.

**Comment by Zarel** [comment]
This means that the offending command is wrong...

**Comment by mia-pi-git** [comment]
Replying here too since it went unnoticed on Discord: it's not wrong, I prefer the validation is done in the shared functions since that way it covers more use-cases. It corrects itself here so that the command doesn't need more code, code that would also need to be elsewhere.

**Comment by Zarel** [comment]
I'm going to merge because this is still a net improvement, but what you're saying doesn't change that the callers are wrong.

**Comment by Zarel** [comment]
Actually I have time to fix this myself.

**Comment by Zarel** [comment]
Thanks!

**Pull request merged**